include/verilated.cpp

```diff
@@ -844,25 +844,25 @@ void _vl_vsformat(std::string& output, const char* formatp, va_list ap) VL_MT_SA
                                    + ((VL_BITISSETLIMIT_W(lwp, lbits, lsb + 2)) ? 4 : 0));
                     }
                     break;
-                case 'u':  // Packed 2-state
-                    output.reserve(output.size() + 4 * VL_WORDS_I(lbits));
-                    for (int i = 0; i < VL_WORDS_I(lbits); ++i) {
-                        output += static_cast<char>((lwp[i]) & 0xff);
-                        output += static_cast<char>((lwp[i] >> 8) & 0xff);
-                        output += static_cast<char>((lwp[i] >> 16) & 0xff);
-                        output += static_cast<char>((lwp[i] >> 24) & 0xff);
-                    }
+                case 'u': { // Packed 2-state
+                    output.reserve(output.size() + VL_BYTES_I(lbits));
+                    for (int bit = 0; bit < 8 * VL_BYTES_I(lbits); bit += 8)
+                        output += static_cast<char>(VL_BITRSHIFT_W(lwp, bit) & 0xff);
```

Comment:
My reading of the standard is the old format was closer as in IEEE raw data is always multiple 32-bit quantities, but please compare to Icarus.